src/credentials/auth.ts

```diff
@@ -902,15 +902,64 @@ export async function createStartUrlPrompter(title: string, ignoreScopes = true)
 }
 
 export async function createBuilderIdConnection(auth: Auth) {
+    const existingConn = await promptLogoutExistingBuilderIdConnection(auth)
+
     const newProfile = createBuilderIdProfile()
-    const existingConn = (await auth.listConnections()).find(isBuilderIdConnection)
     if (existingConn && !hasScopes(existingConn, newProfile.scopes)) {
         return migrateBuilderId(auth, existingConn, newProfile)
     }
 
     return existingConn ?? (await auth.createConnection(newProfile))
 }
 
+/**
+ * Prompts the user to log out of an existing Builder ID connection if it exists.
+ *
+ * @returns The existing Builder ID connection if it exists and the user does not want to log out.
+ *
+ *          Otherwise, undefined, due to user selecting logout or the connection never existing
+ *          in the first place.
+ */
+async function promptLogoutExistingBuilderIdConnection(auth: Auth): Promise<SsoConnection | undefined> {
+    const builderIdConn = (await auth.listConnections()).find(isBuilderIdConnection)
+
+    if (!builderIdConn) {
+        return undefined
+    }
+
+    const items: DataQuickPickItem<string>[] = [
+        {
+            data: 'signout',
+            label: `You already signed in with an ${getIdeProperties().company} Builder ID. Sign out to add another?`,
```

Comment:
Unfortunately the builder-id doesn't have a user-facing "name" , else it would be good to show here too.

```suggestion
            label: `Currently signed in with ${getIdeProperties().company} Builder ID. Sign out to add another?`,
```